services/cannon/src/Cannon/App.hs

```diff
@@ -65,11 +65,10 @@ continue l ws clock k = liftIO $ do
     case result of
         (Left  (Left x)) ->
             let text = client (key2bytes k) . msg (val "read: " +++ show x) in
-            case fromException x of
-                Just ConnectionClosed -> Logger.debug l text
-                _                     -> Logger.warn  l text
-        (Right (Left x)) -> Logger.warn l $
-            client (key2bytes k) . msg (val "write: " +++ show x)
+            Logger.debug l text
```

Comment:
How come no more warning but only at the debug level?

Reply:
there were a lot of warnings about clients closing web sockets untidily.  i think joe and i agreed that they are not very interesting, and just hanging up on a websocket is both legit and unlikely to go away no matter what our moral position on them is.